### PR TITLE
ayugram 5.8.3 (new cask)

### PR DESCRIPTION
**Pull request by moclauq**
**Important:** *Do not tick a checkbox if you haven’t performed its action.* Honesty is indispensable for a smooth review process.

_In the following questions `<cask>` is the token of the cask you're submitting._

After making any changes to a cask, existing or new, verify:

- [x] The submission is for [a stable version](https://docs.brew.sh/Acceptable-Casks#stable-versions) or [documented exception](https://docs.brew.sh/Acceptable-Casks#but-there-is-no-stable-version).
- [x] `brew audit --cask --online <cask>` is error-free.
- [x] `brew style --fix <cask>` reports no offenses.

Additionally, **if adding a new cask**:

- [x] Named the cask according to the [token reference](https://docs.brew.sh/Cask-Cookbook#token-reference).
- [x] Checked the cask was not [already refused](https://github.com/search?q=repo%3AHomebrew%2Fhomebrew-cask+is%3Aclosed+is%3Aunmerged+&type=pullrequests) (add your cask's name to the end of the search field).
- [x] `brew audit --cask --new <cask>` worked successfully.
- [x] `HOMEBREW_NO_INSTALL_FROM_API=1 brew install --cask <cask>` worked successfully.
- [x] `brew uninstall --cask <cask>` worked successfully.

---

### CODE REVIEW

File: Casks/a/ayugram.rb [hunk 1]
```diff
@@ -0,0 +1,16 @@
+cask "ayugram" do
+  version "5.8.3"
+  sha256 "7769048139324700659f73f52b703a65abfb71e6768edc3d3298473565a31849"
+
+  # Build from https://github.com/AyuGram/AyuGramDesktop/releases/download/v5.8.3/AyuGram.dmg
+  url "https://huggingface.co/loxtera/ayu-builds-temp/resolve/main/AyuGram.dmg?download=true"
```
Comment (daeho-ro) [inline Casks/a/ayugram.rb:6]: I think it is better to use the github url, which is versioned and more reliable.

File: Casks/a/ayugram.rb [hunk 2]
```diff
@@ -0,0 +1,16 @@
...
+  name "AyuGram"
+  desc "Is a Telegram client with ghost mode and message history"
```
Comment (daeho-ro) [inline Casks/a/ayugram.rb:8]:
```suggestion
  desc "Telegram client with ghost mode and message history"
```

File: Casks/a/ayugram.rb [hunk 3]
```diff
@@ -0,0 +1,16 @@
...
+  homepage "https://github.com/AyuGram/AyuGramDesktop"
+
+  depends_on macos: ">= :high_sierra"
+
+  app "AyuGram.app", target: "AyuGram Desktop.app"
+
+  zap trash: "~/Library/Application Support/AyuGram Desktop"
```
Comment (daeho-ro) [inline Casks/a/ayugram.rb:15]:
```suggestion
  zap trash: [
    "~/Library/Application Support/AyuGram Desktop",
    "~/Library/Saved Application State/one.ayugram.AyuGramDesktop.savedState",
  ]
```

**Review by daeho-ro — COMMENTED** [review]
Thank you for the PR, @loxetra ! I left some comment and please take a look.

File: Casks/a/ayugram.rb [hunk 4]
```diff
@@ -0,0 +1,16 @@
...
+
+  # Build from https://github.com/AyuGram/AyuGramDesktop/releases/download/v5.8.3/AyuGram.dmg
+  url "https://huggingface.co/loxtera/ayu-builds-temp/resolve/main/AyuGram.dmg?download=true"
+  name "AyuGram"
+  desc "Is a Telegram client with ghost mode and message history"
+  homepage "https://github.com/AyuGram/AyuGramDesktop"
+
+  depends_on macos: ">= :high_sierra"
+
+  app "AyuGram.app", target: "AyuGram Desktop.app"
```
Comment (bevanjkay) [inline Casks/a/ayugram.rb:13]: What is the reason for the change of name here? We likely shouldn't be using `target` here.

**Review by p-linnane — APPROVED** [review]

**Review by khipp — APPROVED** [review]
Thank you for your first contribution to `homebrew-cask`, @loxetra! 🎉
